app/javascript/controllers/click_toggle_controller.js

```diff
@@ -8,29 +8,15 @@ import debounce from "../utils/debounce.js"
 
 export default class extends Controller {
```

Comment:
I implemented your insight about us not needing a "trigger target" since we can just bind to the stimulus action data attribute. In the process, I also managed to eliminate the nested controllers, greatly simplifying things. It turned out quite nice!